test/integration/es/esClient.js

```diff
@@ -54,32 +57,109 @@ describe('esClient', function test() {
       }).catch(done);
     });
 
-    it('should return pharmacies with an alteration that is open', (done) => {
-      const location = { longitude: -1.3601, latitude: 50.9778 };
-      const alterationsOpen = moment('2017-04-15 11:00:00');
-      esClient.getOpenPharmacies(alterationsOpen, location, 25, 2).then((pharmacies) => {
-        expect(pharmacies).to.be.an('array');
-        expect(pharmacies.length).to.equal(2);
-        expect(pharmacies[0].name).to.equal('Boyatt Pharmacy');
-        expect(pharmacies[1].name).to.equal('LloydsPharmacy Inside Sainsbury\'s');
-        done();
-      }).catch(done);
-    });
+    describe('pharmacies with opening hours changed by alterations', () => {
+      const FAQ27DateOfChange = '2016-12-26';
+      const FAQ27 = 'FAQ27';
+      const FCJ43DateOfChange = '2017-04-16';
+      const FCJ43 = 'FCJ43';
 
-    it('should return pharmacies with an alteration that is closed', (done) => {
-      const location = { longitude: -1.3601, latitude: 50.9778 };
-      const alterationsClosed = moment('2017-04-15 08:45:00');
-      esClient.getOpenPharmacies(alterationsClosed, location, 25, 1).then((pharmacies) => {
-        expect(pharmacies).to.be.an('array');
-        expect(pharmacies.length).to.equal(1);
-        expect(pharmacies[0].name).to.equal('Boots');
-        done();
-      }).catch(done);
+      before('set up data', async () => {
+        await esClient.client.update({
```

Comment:
there's repetition of code for updating/getting via the esClient that could be pulled out into a library or functions in the test.

Reply:
Good point.

Reply:
Have you thought of using the `wait_for` option to wait for the update to be applied?

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html

Not sure of the exact syntax in the client but, since the client is just a wrapper around the rest calls, it should be possible.